packages/relay-runtime/network/__tests__/RelayObservable-test.js

```diff
@@ -1444,6 +1444,27 @@ describe('RelayObservable', () => {
       expect(list).toEqual(['error', error]);
     });
 
+    it('Converts an AsyncGenerator to an Observable', done => {
+      const value = {key: 'value'};
+
+      async function* generate() {
+        yield value;
+      }
+
+      // Something else will "run" this before, like the fetchFn
+      const result = generate();
+      const obs = RelayObservable.from(result);
+
+      expect.assertions(1);
```

Comment:
hmm really strange that runAllTimers() isn't working here, as we use that throughout this file for all Promise-related tests without issue. the `obs.subscribe()` call _must_ happen synchronously, it's strange that the earlier alternative i proposed wouldn't work.

Reply:
I think with the promise based stuff, you have the promise handy in scope in order to await, which guarantees the .next's have fired. You can stack awaits.

With generators though, once it's consumed the "iteration", it's consumed. The only way I can think is, to have an async-push-pull mechanism here. Have it such that the subscribe takes place, then emit the next event, kick of timers. Expect, and so on. 

I think what makes this special is that, in all other scenarios the promise is created in scope, whereas generators, although created in scope, the wrapping promise is created inside the sin fn.